lib/python/picongpu/plugins/plot_mpl/phase_space_visualizer.py

```diff
@@ -0,0 +1,158 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017-2018 PIConGPU contributors
+Authors: Sebastian Starke
+License: GPLv3+
+"""
+
+from picongpu.plugins.phase_space import PhaseSpace
+
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+import numpy as np
+
+
+class Visualizer(object):
+    """
+    Class for creating a matplotlib plot of phase space diagrams.
+    """
+
+    def __init__(self, run_directory):
+        """
+        Parameters
+        ----------
+        run_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        if run_directory is None:
+            raise ValueError('The run_directory parameter can not be None!')
+
+        # initialize the data-reader
+        self.phase_space = PhaseSpace(run_directory)
+        self.plt_obj = None
+
+        # for unit-conversion from SI (taken from picongpu readthedocs)
+        self.mu = 1.e6
+        self.e_mc_r = 1. / (9.1e-31 * 2.9979e8)
```

Comment:
if we want beta_gamma here, I have to check if we already specify the species' mass.
also, beta gamma per nucleon for ions would be useful

Reply:
or we just show in units of MeV / c for now...